packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1127,8 +1127,32 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// ```
   ///
   /// It is an error to call this method after the framework calls [dispose].
-  /// You can determine whether it is legal to call this method by checking
+  /// You can determine whether it is legal to call [setState] by checking
   /// whether the [mounted] property is true.
+  ///
+  /// # Performance considerations
+  ///
+  /// Calls to [setState] cause a rebuild of the current widget and,
+  /// potentially, all of its children. This isn't efficient and generally not
+  /// desiderable. A [setState] call should only be made on widgets with small
```

Comment:
This is a bit strong. setState is totally ok to use when the use case fits (e.g. in leave widgets), see also my next comment.

Reply:
To add to this: I have seen very few cases in which setState is actually the root of a performance problem. We shouldn't scare people off using it.